tests/pytests/unit/utils/test_cloud.py

```diff
@@ -220,3 +221,164 @@ def test_winrm_pinnned_version():
             winrm_pkg = pkg_resources.get_distribution("pywinrm")
             assert winrm_pkg.version >= '0.3.0'
     # fmt: on
+
+
+def test_ssh_gateway_arguments_default_alive_args():
+    arguments = ssh_gateway_arguments({"ssh_gateway": "host"})
+    assert "-oServerAliveInterval={}".format(cloud.SERVER_ALIVE_INTERVAL) in arguments
+    assert "-oServerAliveCountMax={}".format(cloud.SERVER_ALIVE_COUNT_MAX) in arguments
+
+
+def test_ssh_gateway_arguments_alive_args():
+    arguments = ssh_gateway_arguments(
+        {
+            "ssh_gateway": "host",
+            "server_alive_interval": cloud.SERVER_ALIVE_INTERVAL + 1,
+            "server_alive_count_max": cloud.SERVER_ALIVE_COUNT_MAX * 2,
+        }
+    )
+    assert (
+        "-oServerAliveInterval={}".format(cloud.SERVER_ALIVE_INTERVAL + 1) in arguments
+    )
+    assert (
+        "-oServerAliveCountMax={}".format(cloud.SERVER_ALIVE_COUNT_MAX * 2) in arguments
+    )
+
+
+@patch("salt.utils.cloud.socket", MagicMock())
```

Comment:
Can we please not use ``patch`` as a decorator?
https://engblog.nextdoor.com/what-your-mocks-do-when-you-aren-t-looking-b278e0d9e201

Reply:
:+1: for this.

`create_autospec` or `autospec=True` is also a good thing (most of the time)

Reply:
Note: instead of decorator, try a fixture:

```
@pytest.fixture
def fake_cloud_socket():
    with patch("salt.utils.cloud.socket", autospec=True):
        yield    # or patch(...) as fake_socket, yeild socke
...

def test_wait_for_port_default_alive_arg(fake_cloud_socket):
    ...
```

Can even do the same thing with the exec_ssh patch

```
def test_wait_for_port_default_alive_args(exec_ssh_cmd, fake_cloud_socket):
    ....
```

using fixtures to remove less relevant/common setup code allows the tests to be *very* narrowly focused :+1: